File: src/pages/exercises/function-triangle-checker/_codes/php/response/src/triangle-checker.php
<?php

function triangleChecker($a, $b, $c)
{
  if ($a >= $b + $c || $b >= $a + $c || $c >= $a + $b)
    return 'none';
  else if ($a === $b && $b === $c)
    return 'equilateral';
  else if ($a !== $b && $b !== $c && $a !== $c)
    return 'scalene';
  else
    return 'isosceles';
}

File: src/pages/exercises/function-triangle-checker/_codes/ruby/response/src/triangle_checker.rb
class TriangleChecker
  def TriangleChecker.compute(a, b, c)
    if (a >= b + c || b >= a + c || c >= a + b)
      :none
    else
      case [a,b,c].uniq.size
      when 1 then :equilateral
      when 2 then :isosceles
      else :scalene
      end
    end
  end
end